homeassistant/components/jellyfin/media_player.py

```diff
@@ -1,363 +1,280 @@
+        ),
+    )
 
 
-class JellyfinMediaPlayerCoordinator(DataUpdateCoordinator):
-    """Bundles data retrieval for sessions form Jellyfin server."""
+class JellyfinMediaPlayer(JellyfinEntity, MediaPlayerEntity):
+    """Represents a Jellyfin Player device."""
 
     def __init__(
         self,
-        hass: HomeAssistant,
-        client: JellyfinClient,
-        user: str,
-        async_add_entities: AddEntitiesCallback,
+        coordinator: JellyfinDataUpdateCoordinator,
+        session_id: str,
+        session_data: dict[str, Any],
     ) -> None:
-        """Initialize the UpdateCoordinator."""
+        """Initialize the Jellyfin Media Player entity."""
         super().__init__(
-            hass, _LOGGER, name="Jellyfin", update_interval=timedelta(seconds=5)
+            coordinator,
+            MediaPlayerEntityDescription(
+                key=session_id,
+            ),
         )
 
-        self._client = client
-        self._hass = hass
-        self.user = user
-        self.sessions: Any | None = None
-
-        self._known: set[str] = set()
-        self._async_add_entities = async_add_entities
+        self.session_id = session_id
+        self.session_data: dict[str, Any] | None = session_data
+        self.device_id: str = session_data["DeviceId"]
+        self.device_name: str = session_data["DeviceName"]
+        self.client_name: str = session_data["Client"]
+        self.app_version: str = session_data["ApplicationVersion"]
 
-    async def _async_update_data(self) -> None:
-        self.sessions = await self._hass.async_add_executor_job(
-            self._client.jellyfin.get_sessions
-        )
+        self.capabilities: dict[str, Any] = session_data["Capabilities"]
+        self.now_playing: dict[str, Any] | None = session_data.get("NowPlayingItem")
+        self.play_state: dict[str, Any] | None = session_data.get("PlayState")
 
-        if not self.sessions:
-            return
+        if self.capabilities.get("SupportsPersistentIdentifier", False):
+            self._attr_device_info = DeviceInfo(
+                identifiers={(DOMAIN, self.device_id)},
+                manufacturer="Jellyfin",
+                model=self.client_name,
+                name=self.device_name,
+                sw_version=self.app_version,
+                via_device=(DOMAIN, coordinator.server_id),
+            )
+        else:
+            self._attr_device_info = None
+            self._attr_has_entity_name = False
+            self._attr_name = self.device_name
 
-        for session in self.sessions:
-            if session["Id"] not in self._known:
-                self._known.add(session["Id"])
-                entity = JellyfinMediaPlayer(self, self._hass, self._client, session)
-                self._async_add_entities([entity], False)
+        self._update_from_session_data()
 
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        self.session_data = (
```

Comment:
This could be easier to read with iterative form.

`if self.coordinator.data is not None and self.session_id in self.coordinator.data` then set all three, else all 3 are None
At least the 2nd and 3rd should be folded into one if, they have the exact same predicate.

Reply:
iirc some of that is due to code coverage flagging the lines and not trying to have to trigger several logic branches that were similiar intent but I'll re-review